dg_prem_proj.bash

```diff
@@ -33,7 +34,7 @@ printf "${GREEN}done.${DEF}\n"
 
 # replace the premiere version with "1"
 printf "updating xml file... "
-sed -i -E "s/\(<Project ObjectID.*Version=\"\).*\(\"\)/\11\2/" "$new_file"
+sed -i "" "s/\(<Project ObjectID.*Version=\"\).*\(\"\)/\11\2/" "$new_file"
```

Comment:
i think this addition would remove the backup in case something goes wrong, I'd rather leave that as-is so that someone doesn't lose all their project on accident

Reply:
You can actually do something like

```bash
UNZIPPED=$(gunzip < $file)
```

where `$file` is the file path. This just unpacks the project to a bash variable, which you can then pipe into `sed`.

Likewise, you can write the downgraded file like this:

```bash
echo "$MODIFIED" | gzip > ${file%.*}-downgraded.prproj
```

which will basically write `input-downgraded.prproj` for any given `input.prproj`.

Wrap the whole thin in a

```bash
for file in "$@"
do
        ...
done
```

and you can insert the script into an Automator workflow plugin for Finder and perform the downgrade by just right clicking any prproj file and using a Quick Action.

Your implementation is a bit different from mine, but I came across yours after having made a similar utility here https://github.com/ocelotsloth/downgrade-premiere-prproj.